docs/DeveloperGuide.md

```diff
@@ -1,34 +1,978 @@
 
-{Describe the target user profile}
+## 2.0 Setting up
+1. Fork the Quotesify repo from [here](https://github.com/AY2021S1-CS2113T-T09-3/tp),
+and clone the fork to your computer.
+2. Open up your IDE (IntelliJ is highly recommended). If you are not at the welcome screen,
+click `File` > `Close Project` to close any existing project.
+3. Set up the correct JDK version for Gradle:
+    1. Click `Configure` > `Project Defaults` > `Project Structure`.
+    2. Click `New…` and find the directory where you saved your JDK.
+4. Click `Import Project`.
+5. Locate the *build.gradle* file and select it.
+6. Click `OK`.
+7. Click `Open as Project`.
+8. Click `OK` to accept all default settings.
+9. To verify the set up, locate the `Quotesify.java` file, right-click it and select `Run Quotesify.main()`.
+If the setup is correct, you should see something like this as shown below:
 
-### Value proposition
+```
+________                __                .__  _____       
+\_____  \  __ __  _____/  |_  ____   _____|__|/ ____\__.__.
+ /  / \  \|  |  \/  _ \   __\/ __ \ /  ___/  \   __<   |  |
+/   \_/.  \  |  (  <_> )  | \  ___/ \___ \|  ||  |  \___  |
+\_____\ \_/____/ \____/|__|  \___  >____  >__||__|  / ____|
+       \__>                      \/     \/          \/    
+Welcome to Quotesify v2.0!
 
-{Describe the value proposition: what problem does it solve?}
+What would you like to do with Quotesify?
+```
 
-## User Stories
+## 3.0 Design
+*`Note: All UML diagrams in this guide are stored in the images/ directory.`*
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+### 3.1 Architecture
+![Sequence Diagram for Architecture](images/SeqDiagram_Architecture.png)
+
+The architecture diagram displayed above describes the high-level design of Quotesify. Below details a brief
+description of each component shown.
+
+#### Main class: `Quotesify.java`
+* On program launch, the Main is responsible for initialising the required components in the correct sequence as well
+as to connect these components to start Quotesify.
+* On shutdown, the Main gracefully terminates the application and its running components.
+
+The other components that make up Quotesify include:
+* **UI**: Text-based User Interface (UI) of the application.
+* **Logic**: Handles the execution of user input commands.
+* **Model**: Stores Quotesify’s data objects in application memory.
+* **Storage**: Stores and accesses program data in the hard disk.
+
+The Sequence Diagram below shows an example of how the components work together upon receiving the command
+`add -b Harry Potter /by JK Rowling`.
+
+![Sequence Diagram for Components](images/SeqDiagram_Components.png)
```

Comment:
![image](https://user-images.githubusercontent.com/7411601/97386926-e6a97080-190f-11eb-9bed-78e851bdfa98.png)

Since we are referring to an object, shall we add `:` before the class name? (e.g. `:Ui` instead of `Ui`)
Should we omit the storage and replace `storage.save()` to `save()` since we are aware that `save()` belong to the `storage` class via the diagram?